OSB-45: allowing modifying color theme and logo

sniedzielski opened the pull request:
TICKET: https://openimis.atlassian.net/browse/OSB-45

related PR: https://github.com/openimis/openimis-fe-core_js/pull/271

sniedzielski committed:
OSB-45: allowing modifying color theme

sniedzielski committed:
OSB-45: added processing logo update or keeping default one

sniedzielski committed:
OSB-45: make the white logo optional

sniedzielski committed:
OSB-45: added README how to configure logo and theme

sniedzielski committed:
OSB-45: changed icon for 'logo' section in README

delcroip commented on src/helpers/theme.js:
```diff
@@ -1,223 +1,248 @@
+      right: 8,
+      zIndex: 2000,
       display: "flex",
-      flexWrap: "nowrap",
```
any reason to remove nowrap ?

sniedzielski replied:
The same as prev comment. This is still under 'buttonContainer' as it is on develop. Git 'diff' shows this in an inappropriate misleading form.

delcroip commented on src/helpers/theme.js:
```diff
@@ -1,223 +1,248 @@
-  form: {
-    spacing: 10,
-  },
-  formTable: {
```
why we hadd formTable{ paper and now paper{ paper ?

sniedzielski replied:
![Screenshot from 2025-05-08 09-58-50](https://github.com/user-attachments/assets/e2ab4db3-94e8-4788-8c46-717958235884)
That's normal. You can see that on develop there is the same situation. The same with formTable
You can take a look into https://github.com/openimis/openimis-fe_js/blob/2e887c7e817c7bf583795488700fadff7f6b73b7/src/helpers/theme.js and compare with develop https://github.com/openimis/openimis-fe_js/blob/develop/src/helpers/theme.js

delcroip reviewed (commented):
it seems that the theme.js structure changed

weilu commented on README.md:
```diff
@@ -294,3 +294,90 @@ Conclusions:
+#### 🖼️ Logo Configuration (`logo`)
+
+Add additional property 
+under `fe-core` configuration in `moduleConfiguration`. Logos can also be configured via base64-encoded images. This allows the UI to show custom branding without bundling new static files.
```
Might be good to indicate the supported formats: png, jpeg, svg+xml

weilu commented on src/index.js:
```diff
@@ -94,6 +100,7 @@ const AppContainer = () => {
                 localesManager={localesManager}
                 messages={messages_ref}
                 logo={logo}
+                {...(whiteLogo !== defaultLogo && { whiteLogo })}
```
@sniedzielski I saw https://github.com/openimis/openimis-fe-core_js/pull/271/files has handling of switching between logo and whiteLogo – wouldn't it be cleaner to bring that logic in here to keep all the logo switching/determination logic in one place and only pass `logo={logo}` to App? essentially just replace the logo & whiteLogo definition lines with below and get rid of `whiteLogo`

```
const logo = config?.["fe-core"]?.logo?.white ? getConfiguredLogo(appState.config, "white") : getConfiguredLogo(appState.config);
```

sniedzielski committed:
OSB-45: added changes related to white logo and text processing next to logo

sniedzielski commented:
@weilu I decided to remove `whiteLogo` as you proposed and keep only logo (I changed too in 2nd PR). Instead of this I added additional flag `disableTextLogo` (by default False) which is responsible for removing text 'openIMIS' next to the logo for example
![Screenshot from 2025-05-08 12-25-48](https://github.com/user-attachments/assets/1a151b03-9d04-4218-8415-57486c73d1ca)
![Screenshot from 2025-05-08 12-26-55](https://github.com/user-attachments/assets/f76679c1-cd42-43de-ac3d-7c11e61466d9)

When there is default or False, the text should be displayed next to logo as below:
![Screenshot from 2025-05-08 12-26-09](https://github.com/user-attachments/assets/8d96a81d-fe4f-4d26-bab7-1668f5e46770)

CC @delcroip 

FYI @delcroip @weilu I resolved all comments you added. Please re-check and merge if you agree with that solution.
